src/box/lua/net_box.lua

```diff
@@ -390,6 +391,24 @@ local function new_sm(uri, opts)
                     })
                 end
             end
+            if not opts.fetch_schema and features.space_and_index_names then
+                remote.space = setmetatable({_remote = remote}, {
```

Comment:
Theoretically, there may be a space named `_remote`, in which case it'll be masked by the private field. We could avoid that by binding `remote` to the `__index` function. However, this would require creation of a new `__index` function per each connection (which you already do). I'm not sure it's okay. We should ask @Totktonada.

Reply:
AFAIC, spaces with ’_’ prefix are reserved to be system spaces, so that cannot happen.

Reply:
Theoretically, we can create a system space with this name (`_remote`) either in core or in some system module (e.g. vshard). We could fix this by using captured (bound) values in `__index`.

Reply:
Then AFAIC we won't be able to make the connection a weak reference (see discussion above).

Reply:
Fixed, now `space` and `index` table work as registers.